lib/ansible/module_utils/common/text/formatters.py

```diff
@@ -53,7 +65,8 @@ def human_to_bytes(number, default_unit=None, isbits=False):
         The function expects 'b' (lowercase) as a bit identifier, e.g. 'Mb'/'Kb'/etc.
         if 'MB'/'KB'/... is passed, the ValueError will be rased.
     """
-    m = re.search(r'^\s*(\d*\.?\d*)\s*([A-Za-z]+)?', str(number), flags=re.IGNORECASE)
+    m = re.search(r'^([0-9]*\.?[0-9]+)(?:\s*([A-Za-z]+))?\s*$', str(number), flags=re.IGNORECASE)
```

Comment:
I would replace `[0-9]` with `\d` ?

Reply:
@Akasurde the problem with using \d is that it will allow for things like ᭔ MB(which is probabaly not what we want), since \d will match for any Unicode digit. Hence the reason for [0-9].

Reply:
`[A-Za-z]` is redundant with `re.IGNORECASE`

Reply:
@yctomwang Can you please address @flowerysong's comment? rest LGTM

Reply:
@flowerysong Hi, the [A-Za-z] is needed for matching, however with the re.IGNORECASE, the pattern could be simplifed to [a-z] since the flag makes it case non sensitive. Without the [a-z] or [A-Za-z] the pattern wont work, the re.IGNORECASE flag only affects case sensitivity but does not introduce any character classes by itself. I tried make the pattern like r'^([0-9]*\.?[0-9]+)(?:\s*())?\s*$', without the [A-Za-z] however its starting to fail for things like '1YB'.

Reply:
@flowerysong I am happy to modify the pattern to from [A-Za-z] to [a-z], if this address your concern. Thanks for reviewing this PR!

Reply:
I think it's slightly more readable to remove the flag, but I don't have a strong preference either way.

Reply:
@Akasurde addressed the comment and commited, please review and let me know if there are any problems! Thanks again!